### ISSUE TITLE
Testing utility

### ISSUE THREAD

**Issue by eladb** [awslabs/cdk8s#169]
We would like to offer utilities for unit testing in the `cdk8s.Testing` class.

Here are some examples of useful assertions:

1. Assert that a chart synthesizes a manifest with a specific API object of a specific kind and optionally: `apiVersion`, a subset of properties (or exact match)
2. assert count of API objects from a certain kind
3. Assert that a chart synthesizes a manifest that matches a snapshot stored in a file (would be nice to make it easy to update the snapshot by e.g. setting an env. variable).

These functions should be testing-framework agnostic (and naturally jsii-able) so they can be used in all languages. If people want to create wrappers/plugins for specific frameworks, it should be easy to wrap these utilities.

**Comment by manparvesh** [comment]
Hey @eladb can I work on this issue? I'm really interested in contributing to this project.

**Comment by eladb** [comment]
Yes, absolutely. You can find inspiration in the how the [AWS CDK testing](https://docs.aws.amazon.com/cdk/latest/guide/testing.html) works and the [@aws-cdk/assert](https://www.npmjs.com/package/@aws-cdk/assert) module. I was wondering if perhaps we should actually write this as a general purpose JSON assertion library instead of cdk8s specific.

Would you be able to write a quick [RFC](https://github.com/awslabs/cdk8s/blob/master/CONTRIBUTING.md#rfcs) that plots your plan for this so we will make sure we are on the same page?

**Comment by wuqunfei** [comment]
Is anyone can share detail implementation direction we can go? this feature and CDK pipeline feature can make difference between the helm and programming language. I am join to contriube on it.
